framework/src/org/checkerframework/framework/type/DefaultTypeHierarchy.java

```diff
@@ -920,6 +920,15 @@ public Boolean visitWildcard_Array(
     @Override
     public Boolean visitWildcard_Declared(
             AnnotatedWildcardType subtype, AnnotatedDeclaredType supertype, VisitHistory visited) {
+        if (subtype.isUninferredTypeArgument() && supertype.getTypeArguments().isEmpty()) {
```

Comment:
I don't understand this check and/or the explanation. If the subtype is a wildcard, why does it matter whether the supertype has type arguments or not? The wildcard is not an argument for any of those type arguments, so the condition seems irrelevant.